pkg/vault/controller.go

```diff
@@ -40,17 +42,38 @@ func NewController(vconfig *vaultapi.Config, kconfig *rest.Config) (kube.SecretC
 	}
 
 	return &controller{
-		vclient: vclient,
-		kclient: kclient,
+		vclient:         vclient,
+		kclient:         kclient,
+		namespacePrefix: namespacePrefix,
 	}, nil
 }
 
+func pathAllowed(path string, prefix string, namespace string) bool {
+	// If this secret is outside the prefix it is always allowed
+	if !strings.HasPrefix(path, prefix) {
+		return true
+	}
+
+	// If the path starts with the namespace prefix + the namespace then it is allowed
+	if strings.HasPrefix(path, prefix+namespace+"/") {
```

Comment:
nit: maybe we could use `fmt.Sprintf` to concatenate this strings. Untested suggestion (note that the suggestion assumes there's no trailing `/` in the prefix):

```suggestion
	if strings.HasPrefix(path, fmt.Sprintf("%s/%s/", prefix, namespace)) {
```

Reply:
`fmt.Sprintf` would be a nice fit if there was more complex formatting going on. It's debatable about which one is more readable (I prefer plain concatenation but there is no official style guide answer to this and everyone has different opinions). The main reason for choosing to use `+` is that it is around 10x faster than `fmt.Sprintf`.

From a project consistency point of view there is currently no usage of `fmt.Sprintf` and 1 usage of basic concatenation.

Reply:
I knew you were going to bring `speed` as a point in favor of `+` over `Sprintf`. Trust me, speed is not always the best choice. I prefixed my comment as nit, which means I'm ok with the current implementation if you don't want to change it.

> From a project consistency point of view there is currently no usage of fmt.Sprintf and 1 usage of basic concatenation.

I'm a huge fan of consistency but, often enough, consistency needs to be revisited.